crates/valence_boss_bar/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+#![doc = include_str!("../README.md")]
+#![allow(clippy::type_complexity)]
+#![deny(
+    rustdoc::broken_intra_doc_links,
+    rustdoc::private_intra_doc_links,
+    rustdoc::missing_crate_level_docs,
+    rustdoc::invalid_codeblock_attributes,
+    rustdoc::invalid_rust_codeblocks,
+    rustdoc::bare_urls,
+    rustdoc::invalid_html_tags
+)]
+#![warn(
+    trivial_casts,
+    trivial_numeric_casts,
+    unused_lifetimes,
+    unused_import_braces,
+    unreachable_pub,
+    clippy::dbg_macro
+)]
+
+use std::borrow::Cow;
+
+use bevy_app::CoreSet::PostUpdate;
+use bevy_app::Plugin;
+use bevy_ecs::prelude::Entity;
+use bevy_ecs::query::{Added, Changed, With};
+use bevy_ecs::schedule::IntoSystemConfigs;
+use bevy_ecs::system::Query;
+use packet::{BossBarAction, BossBarS2c};
+use valence_client::{Client, FlushPacketsSet};
+use valence_core::despawn::Despawned;
+use valence_core::protocol::encode::WritePacket;
+use valence_core::uuid::UniqueId;
+
+mod components;
+pub use components::*;
+
+pub mod packet;
+
+pub struct BossBarPlugin;
+
+impl Plugin for BossBarPlugin {
+    fn build(&self, app: &mut bevy_app::App) {
+        app.add_systems(
+            (
+                boss_bar_title_update,
+                boss_bar_health_update,
+                boss_bar_style_update,
+                boss_bar_flags_update,
+                boss_bar_viewers_update,
+                boss_bar_despawn,
+                client_disconnection,
```

Comment:
`client_disconnection` should be ordered before `boss_bar_viewers_update` so the updates are done on the same tick.

Reply:
Oh I see, I added that `client_disconnection.before(boss_bar_viewers_update)`, so sure that's how I should do though

Reply:
That's correct. Just need to `cargo fmt` and we can merge this.